parity: 2.5.6 -> 2.5.7, parity-beta: 2.6.1 -> 2.6.2

xrelkd opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
Update `parity` and `parity-beta` to the latest releases.
- https://github.com/paritytech/parity-ethereum/releases/tag/v2.5.7
- https://github.com/paritytech/parity-ethereum/releases/tag/v2.6.2
- https://www.parity.io/parity-ethereum-security-release-29-8-19/

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @akru

akru reviewed (approved):
Thank you for your work!

xrelkd commented:
@akru You're welcome!
Hi, @worldofpeace @marsam could you review this PR? Thank you very much!

worldofpeace commented on pkgs/applications/blockchains/parity/parity.nix:
```diff
@@ -26,10 +26,7 @@ rustPlatform.buildRustPackage rec {
-    pkgconfig cmake perl
-    systemd.lib systemd.dev openssl openssl.dev
-  ];
+  buildInputs = [ cmake openssl perl pkgconfig systemd ];
```
Are we sure we need `cmake` and `pkgconfig` as `buildInputs`? Those are typically only needed during the build so they're `nativeBuildInputs`.

worldofpeace replied:
Also where is perl needed https://github.com/paritytech/parity-ethereum/search?q=perl&type=Code ?

xrelkd committed:
parity: Update buildInputs and nativeBuildInputs

Move cmake and pkgconfig from buildInputs to nativeBuildInputs
Remove unneeded perl from buildInputs

xrelkd committed:
parity: 2.5.6 -> 2.5.7

xrelkd committed:
parity-beta: 2.6.1 -> 2.6.2

xrelkd commented:
@worldofpeace Thanks for the review!
`buildInputs` and `nativeBuildInputs` have been updated.

@GrahamcOfBorg build parity parity-beta

xrelkd commented:
Hi @worldofpeace, is there any other improvement I can do for this PR?

worldofpeace commented:
> Hi @worldofpeace, is there any other improvement I can do for this PR?

Seeing if it can be integrated now. Thanks for the reminder.

worldofpeace reviewed (approved):
LGTM. Still builds.
